### ISSUE TITLE
To the future ! (= compatibility issues with modern IRC clients)

### ISSUE THREAD

**Issue by Fnux** [Fnux/hanabi#11]
Some IRC clients doesn't play well with Hanabi since they depend on messages specified by [RFC2812](https://tools.ietf.org/html/rfc2812) while Hanabi (currently) only implement [RFC1459](https://tools.ietf.org/html/rfc1459).

We should at least implement modern "Connection Registration" as specified/explained on [https://modern.ircdocs.horse/index.html#connection-registration](https://modern.ircdocs.horse/index.html#connection-registration) :
> * If the server supports capability negotiation, the CAP command suspends the registration process and immediately starts the capability negotiation process. CAP LS 302 means that the client supports version 302 of client capability negotiation. The capability negotiation process is resumed when the client sends CAP END to the server.
> * The PASS command is not required for the connection to be registered, but if included it MUST precede the latter of the NICK and USER commands.
> * The NICK and USER commands are used to set the user’s nickname, username and “real name”. Unless the registration is suspended by a CAP negotiation, these commands will end the registration process.
> * The client should request advertised capabilities it wishes to enable here.
> * If the client supports SASL authentication and wishes to authenticate with the server, it should attempt this after a successful CAP ACK of the sasl capability is received and while registration is suspended.
> * If the server support capability negotiation, CAP END will end the negotiation period and resume the registration.

> Upon successful completion of the registration process, the server MUST send, in this order, the RPL_WELCOME (001), RPL_YOURHOST (002), RPL_CREATED (003), RPL_MYINFO (004), and at least one RPL_ISUPPORT (005) numeric to the client. The server SHOULD then respond as though the client sent the LUSERS command and return the appropriate numerics. If the user has client modes set on them automatically upon joining the network, the server SHOULD send the client the RPL_UMODEIS (221) reply. The server MAY send other numerics and messages. The server MUST then respond as though the client sent it the MOTD command, i.e. it must send either the successful Message of the Day numerics or the ERR_NOMOTD numeric.

@tigertoes, do you plan to work on it ? We will step on each other if we work on the same part.

**Comment by Fnux** [comment]
I will add `RPL_WELCOME (001), RPL_YOURHOST (002), RPL_CREATED (003), RPL_MYINFO (004)` to the "greeting" either tonight or tomorrow.

**Comment by Fnux** [comment]
See 370d8b2fbd6e45a43df2b1d51b12596ebe523f90 (*Add RPL_{WELCOME, YOURHOST, CREATED, MYINFO} to the greeting sequence*).

**Comment by tigertoes** [comment]
@fnux Thank you for picking this up in my absence, It appears current HEAD (6e16dbeb) connects, channels work however irssi is failing assertions, with `-!- Irssi: critical nicklist_set_host: assertion 'host != NULL' failed` being thrown, I suspect it's happening [in this call](https://github.com/irssi/irssi/blob/master/src/core/nicklist.c#L96). WHOIS on the client side shows the host as "*". I presume this is just something not yet added?

**Comment by Fnux** [comment]
You're right, the current implementation of is incomplete : https://github.com/Fnux/hanabi/blob/master/lib/hanabi/irc/handler.ex#L519.

PS: I don't think I will have enough time to work on Hanabi during the next 2 weeks.

**Comment by tigertoes** [comment]
Thank you for what you've done so far, if I find some time I will see what I can do towards contributing further.
